core/src/test/java/org/elasticsearch/common/settings/SettingTests.java

```diff
@@ -154,6 +154,20 @@ public void testUpdaterIsIsolated() {
         assertNull(ab2.get());
     }
 
+    public void testDeprecatedSetting() {
+        final Setting<Boolean> deprecatedSetting = Setting.boolSetting("deprecated.foo.bar", false, Property.Deprecated);
+        final Settings settings = Settings.builder().put("deprecated.foo.bar", true).build();
+        final int iterations = randomIntBetween(1, 128);
```

Comment:
In fact, your comment made me realize that we should test zero too so I pushed 8a412fab24a0e884cdbdb3efa2acb366921944da.